feature-libs/customer-ticketing/components/customer-ticketing-create/customer-ticketing-create-dialog/customer-ticketing-create-dialog.component.ts

```diff
@@ -1,29 +1,121 @@
-import { Component, OnInit } from '@angular/core';
-import { AssociatedObject, Category } from '@spartacus/customer-ticketing/root';
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import {
+  AssociatedObject,
+  Category,
+  TicketStarter,
+} from '@spartacus/customer-ticketing/root';
 import { FormUtils } from '@spartacus/storefront';
-import { Observable } from 'rxjs';
+import { Observable, Subscription } from 'rxjs';
 import { CustomerTicketingDialogComponent } from '../../shared/customer-ticketing-dialog/customer-ticketing-dialog.component';
 @Component({
   selector: 'cx-customer-ticketing-create-dialog',
   templateUrl: './customer-ticketing-create-dialog.component.html',
 })
 export class CustomerTicketingCreateDialogComponent
   extends CustomerTicketingDialogComponent
-  implements OnInit
+  implements OnInit, OnDestroy
 {
   ticketCategories$: Observable<Category[]> =
     this.customerTicketingFacade.getTicketCategories();
   ticketAssociatedObjects$: Observable<AssociatedObject[]> =
     this.customerTicketingFacade.getTicketAssociatedObjects();
+  subscription: Subscription;
+  ticketStarter: TicketStarter;
+  ticketCategories: Array<Category>;
+  ticketAssociatedObjects: Array<AssociatedObject>;
+
+  @Input()
+  selectedCategory: Category;
+
+  @Input()
+  selectedAssociatedObject: AssociatedObject;
+
+  protected getCreateTicketPayload(form: FormGroup): TicketStarter {
+    return {
+      message: form?.get('message')?.value,
+      subject: form?.get('subject')?.value,
+      associatedTo: this.selectedAssociatedObject,
+      ticketCategory: this.selectedCategory,
+    };
+  }
 
   ngOnInit(): void {
     this.buildForm();
   }
 
+  protected buildForm(): void {
+    const form = new FormGroup({});
+    form.setControl(
+      'subject',
+      new FormControl('', [
+        Validators.required,
+        Validators.maxLength(this.inputCharactersForSubject),
+      ])
+    );
+    form.setControl(
+      'ticketCategory',
+      new FormControl('', [Validators.required])
+    );
+    form.setControl('associatedTo', new FormControl());
+    form.setControl(
+      'message',
+      new FormControl('', [
+        Validators.required,
+        Validators.maxLength(this.inputCharactersLimit),
+      ])
+    );
+    form.setControl(
+      'file',
+      new FormControl('', [
+        this.filesFormValidators.maxSize(this.maxSize),
+        this.filesFormValidators.maxEntries(this.maxEntries),
```

Comment:
I am getting an exception that is thrown for an empty file, should we also add file not supported for empty files?